.github/workflows/generate-pinval.yaml

```diff
@@ -83,23 +95,46 @@ jobs:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
           aws-region: us-east-1
 
-      - name: Collect township codes (or fallback)
-        id: set-matrix
+      - name: Resolve model metadata
+        id: resolve-metadata
         run: |
-          python3 scripts/generate_pinval/list_townships.py \
+          # Parse PIN list to an array so we can safely pass it into this
+          # script, which accepts multiple space-delimited PINs
+          read -ra PINS <<< "${{ inputs.pins }}"
```

Comment:
Note that [space is included in the default value for the Unix internal field separator (IFS)](https://bash.cyberciti.biz/guide/$IFS), so I've removed the `IFS` assignment in adapting the following line to the one you see here:

https://github.com/ccao-data/pinval/blob/9e3972bebcb25b0c3025f568de67c0838ff5443b/.github/workflows/generate-pinval.yaml#L163